### PR TITLE
Rf07 tc001 tc ivan misericordia 02

### PR DESCRIPTION
**Pull request by ivanMise**
RF07-TC-001 UI >Registro de Equipamiento > Verificar cabecera de lista de Equipamientos (Nombre, Acciones).

image

image

### CODE REVIEW

File: src/main/java/org/umssdiplo/automationv01/core/managepage/Equipment/ListEquipment.java [hunk 1]
```diff
@@ -13,11 +13,21 @@
     @FindBy(className = "mat-table")
     private WebElement tableEquipments;
 
+    @FindBy(xpath = "/html/body/ssi-root/ssi-equipment/div/div[3]/div/div/mat-table/mat-header-row/mat-header-cell[1]")
+    private WebElement listNameRow;
```
Comment (ctola91) [inline src/main/java/org/umssdiplo/automationv01/core/managepage/Equipment/ListEquipment.java]: Change name of element

File: src/main/java/org/umssdiplo/automationv01/core/managepage/Equipment/ListEquipment.java [hunk 2]
```diff
@@ -13,11 +13,21 @@
...
+
+    @FindBy(xpath = "/html/body/ssi-root/ssi-equipment/div/div[3]/div/div/mat-table/mat-header-row/mat-header-cell[2]")
+    private WebElement listActionsRow;
```
Comment (ctola91) [inline src/main/java/org/umssdiplo/automationv01/core/managepage/Equipment/ListEquipment.java]: Change name of element

File: src/test/java/org/umssdiplo/automationv01/stepdefinitionproject/StepsDefinitionSSID.java [hunk 3]
```diff
@@ -173,12 +173,14 @@ public void validarQuePaginadorSeMuestraDebajoDeLaTablaListaDeCargo() throws Thr
     }
     //End Positions
 
+    //BEGIN Equipment
```
Comment (ctola91) [inline src/test/java/org/umssdiplo/automationv01/stepdefinitionproject/StepsDefinitionSSID.java:197]: Delete unnecesary comments

File: src/test/java/org/umssdiplo/automationv01/stepdefinitionproject/StepsDefinitionSSID.java [hunk 4]
```diff
@@ -173,12 +173,14 @@ public void validarQuePaginadorSeMuestraDebajoDeLaTablaListaDeCargo() throws Thr
...
+
     @And("^seleccionar menu 'Equipamiento' en la pagina 'Menu Principal'$")
     public void menuEquipamientoEstaSeleccionado() throws Throwable {
         menuEquipamiento = menu.selectEquipmentMenu();
     }
 
-    @And("^Seleccionar submenu 'Equipamiento' en menu 'Equipamiento'$")
+    @And("^seleccionar submenu 'Equipamiento' en menu 'Equipamiento'$")
```
Comment (cmtp) [inline src/test/java/org/umssdiplo/automationv01/stepdefinitionproject/StepsDefinitionSSID.java]: Uppercase

**Review by ctola91 — CHANGES_REQUESTED** [review]

File: src/main/java/org/umssdiplo/automationv01/core/managepage/Equipment/ListEquipment.java [hunk 5]
```diff
@@ -8,26 +8,26 @@
...
+    private WebElement buttonCreateEquipment;
 
     @FindBy(className = "mat-table")
     private WebElement tableEquipments;
 
     @FindBy(xpath = "/html/body/ssi-root/ssi-equipment/div/div[3]/div/div/mat-table/mat-header-row/mat-header-cell[1]")
-    private WebElement listNameRow;
+    private WebElement columnListName;
 
     @FindBy(xpath = "/html/body/ssi-root/ssi-equipment/div/div[3]/div/div/mat-table/mat-header-row/mat-header-cell[2]")
```
Comment (ctola91) [inline src/main/java/org/umssdiplo/automationv01/core/managepage/Equipment/ListEquipment.java]: Please dont use a large xpath

File: src/main/java/org/umssdiplo/automationv01/core/managepage/Equipment/ListEquipment.java [hunk 6]
```diff
@@ -8,26 +8,26 @@
 public class ListEquipment extends BasePage {
 
     @FindBy(id = "crearEquipment")
-    private WebElement btnCrearEquipment;
+    private WebElement buttonCreateEquipment;
 
     @FindBy(className = "mat-table")
     private WebElement tableEquipments;
 
     @FindBy(xpath = "/html/body/ssi-root/ssi-equipment/div/div[3]/div/div/mat-table/mat-header-row/mat-header-cell[1]")
```
Comment (ctola91) [inline src/main/java/org/umssdiplo/automationv01/core/managepage/Equipment/ListEquipment.java]: Please dont use a large xpath

File: src/main/java/org/umssdiplo/automationv01/core/managepage/Equipment/ListEquipment.java (hunk 5 above)
Comment (ctola91) [reply src/main/java/org/umssdiplo/automationv01/core/managepage/Equipment/ListEquipment.java]: Use contains maybe to fix this

**Review by ctola91 — CHANGES_REQUESTED** [review]

File: src/main/java/org/umssdiplo/automationv01/core/managepage/Equipment/ListEquipment.java [hunk 7]
```diff
@@ -8,16 +8,26 @@
 public class ListEquipment extends BasePage {
 
     @FindBy(id = "crearEquipment")
-    private WebElement btnCrearEquipment;
+    private WebElement buttonCreateEquipment;
 
     @FindBy(className = "mat-table")
     private WebElement tableEquipments;
 
+    @FindBy(xpath = "/html/body/ssi-root/ssi-equipment/div/div[3]/div/div/mat-table/mat-header-row/mat-header-cell[1]")
```
Comment (marbusjim) [inline src/main/java/org/umssdiplo/automationv01/core/managepage/Equipment/ListEquipment.java]: Use a short xpath

File: src/main/java/org/umssdiplo/automationv01/core/managepage/Equipment/ListEquipment.java [hunk 8]
```diff
@@ -8,16 +8,26 @@
...
+    private WebElement columnListName;
+
+    @FindBy(xpath = "/html/body/ssi-root/ssi-equipment/div/div[3]/div/div/mat-table/mat-header-row/mat-header-cell[2]")
```
Comment (marbusjim) [inline src/main/java/org/umssdiplo/automationv01/core/managepage/Equipment/ListEquipment.java]: Use a short xpath

File: src/test/java/org/umssdiplo/automationv01/stepdefinitionproject/StepsDefinitionSSID.java [hunk 9]
```diff
@@ -187,4 +188,10 @@ public void seleccionarSubMenuEquipamiento() throws Throwable {
     public void validarListaDeEquipamientos() throws Throwable {
         Assert.assertTrue(listEquipment.isEquipmentListVisible(), String.format(ErrorMessage.ERROR_MESSAGE_ELEMENT_VISIBLE, "Equipments title"));
     }
+
+    @Then("^verificar cabecera 'Nombre' y 'Acciones' de la pagina 'Lista de Equipamientos' esten cargados$")
+    public void verificarCabeceraListaEquipamientos() throws Throwable {
+        Assert.assertTrue(listEquipment.checkHeaderListEquipment(), String.format(ErrorMessage.ERROR_MESSAGE_ELEMENT_VISIBLE, "Equiment title"));
```
Comment (marbusjim) [inline src/test/java/org/umssdiplo/automationv01/stepdefinitionproject/StepsDefinitionSSID.java]: you need verify headers in your list according your step, currently you only validate the list in page

**Review by marbusjim — COMMENTED** [review]
please review comments

File: src/test/java/org/umssdiplo/automationv01/stepdefinitionproject/StepsDefinitionSSID.java [hunk 10]
```diff
@@ -204,6 +210,12 @@ public void validarListaDeEquipamientos() throws Throwable {
         Assert.assertTrue(listEquipment.isEquipmentListVisible(), String.format(ErrorMessage.ERROR_MESSAGE_ELEMENT_VISIBLE, "Equipments title"));
     }
 
+    @Then("^verificar cabecera 'Nombre' y 'Acciones' de la pagina 'Lista de Equipamientos' esten cargados$")
```
Comment (marbusjim) [inline src/test/java/org/umssdiplo/automationv01/stepdefinitionproject/StepsDefinitionSSID.java]: if you use this verification, you need add two validations (validationNameHeader and validationActionHeader)

**Review by ctola91 — APPROVED** [review]

**Review by marbusjim — APPROVED** [review]
Approve

**Pull request merged**